### PR TITLE
Add Welsh and Gaelic machine translation providers

### PR DESCRIPTION
**Pull request by phlax**
This PR adds providers for the following backend translation providers

- Gaelic to Gaelic:
   https://github.com/kscanne/caighdean/blob/master/API.md
- English to Welsh:
  http://techiaith.cymru/api/translation/?lang=en
  

cc @kscanne @techiaith

### CODE REVIEW

File: pootle/static/js/editor/mt/providers/CaighdeanTranslate.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+class CaighdeanTranslate extends MTProvider {
+
+  constructor(apiKey) {
+    super({
+      apiKey,
+      name: 'caighdean-translate',
+      displayName: 'Caighdean Translate',
+      url: "http://borel.slu.edu/cgi-bin/seirbhis3.cgi",
+	supportedLanguages: ['gd', 'gv', 'ga_IE', 'ga-IE'],
```
Comment (unho) [inline pootle/static/js/editor/mt/providers/CaighdeanTranslate.js]: This is not properly indented. Probably this and other indents are causing Travis to fail.

File: pootle/static/js/editor/mt/providers/CaighdeanTranslate.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    });
+  }
+
+  getRequestBody(opts) {
+    return {
+      teacs: opts.text,
+      foinse: opts.sourceLanguage,
```
Comment (unho) [inline pootle/static/js/editor/mt/providers/CaighdeanTranslate.js]: What are `foinse` and `teacs`? Shouldn't these be `q` and `source`?

Comment (unho) [inline pootle/static/js/editor/mt/providers/CaighdeanTranslate.js]: Forget about that. Stupid question.

File: pootle/static/js/editor/mt/providers/CaighdeanTranslate.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    };
+  }
+    handleSuccess(response) {
+		var i, text = '';
+	for (i = 0; i < response.length; i++) {
```
Comment (unho) [inline pootle/static/js/editor/mt/providers/CaighdeanTranslate.js]: Nit: I find it cleaner to use `for (let i = 0;`.

**Comment by unho** [comment]
Generally the styling for the JavaScript code is a mess, despite the changes look good. Will review again when Travis passes.

**Comment by julen** [comment]
It'd be good to fully add each provider on its own commit, including setting changes and icon.

Other than that this looks good to me :+1:, thanks for taking care of it!

**Comment by unho** [comment]
I was about to say the same, a single commit for adding everything required for each MT provider.

Other than that lgtm.

**Comment by techiaith** [comment]
I've just emailed Ryan with a more suitable API key for Mozilla and other translation work via Pootle. Diolch yn fawr!

**Pull request merged**